### PR TITLE
draw layer data on map by source type

### PR DESCRIPTION
**Pull request by Blaussh**
… plugin for OSM layers

### CODE REVIEW

File: src/app/@ansyn/plugins/ansyn-plugins.module.ts [hunk 1]
```diff
@@ -40,7 +42,9 @@ import { CesiumMap } from '@ansyn/plugins/cesium/cesium-map/cesium-map';
 				GoToVisualizer,
 				PinPointVisualizer,
 				MouseShadowVisualizer,
-				PolygonSearchVisualizer
+				PolygonSearchVisualizer,
+				OpenlayersOsmLayersPlugin,
+				OpenlayersGeoJsonLayersPlugin
```
[the remark this answers is not in the document]
Comment (Blaussh) [reply src/app/@ansyn/plugins/ansyn-plugins.module.ts]: 👍

File: src/app/@ansyn/plugins/openlayers/layers/openlayers-geoJson-layers.plugin.ts [hunk 2]
```diff
@@ -0,0 +1,77 @@
+import { OpenLayersMap } from '@ansyn/plugins/openlayers/open-layers-map/openlayers-map/openlayers-map';
```
[the remark this answers is not in the document]
Comment (Blaussh) [reply src/app/@ansyn/plugins/openlayers/layers/openlayers-geoJson-layers.plugin.ts]: 👍

File: src/app/@ansyn/map-facade/reducers/map.reducer.ts [hunk 3]
```diff
@@ -113,12 +113,15 @@ export function MapReducer(state: IMapState = initialMapState, action: MapAction
 
 			return { ...state, pendingOverlays: pendingOverlaysClone };
 
-		case CoreActionTypes.TOGGLE_MAP_LAYERS:
-			const toggledMap = state.mapsList.find(map => map.id === action.payload.mapId);
+		case CoreActionTypes.TOGGLE_MAP_LAYERS: {
+			const mapsList = [ ...state.mapsList ];
+			const toggledMap = mapsList.find(map => map.id === action.payload.mapId);
 			if (toggledMap) {
 				toggledMap.flags.layers = !toggledMap.flags.layers;
```
[the remark this answers is not in the document]
Comment (Blaussh) [reply src/app/@ansyn/map-facade/reducers/map.reducer.ts]: 👍

File: src/app/@ansyn/plugins/openlayers/layers/openlayers-geoJson-layers.plugin.ts [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+								this.clearEntities();
+								observer.next(true);
+							});
+						}
+					});
+				return forkJoin(obsList);
+			})
+		);
+
+	isGeoJsonLayer(layer: ILayer) {
```
[the remark this answers is not in the document]
Comment (Blaussh) [reply src/app/@ansyn/plugins/openlayers/layers/openlayers-geoJson-layers.plugin.ts]: 👍

File: src/app/@ansyn/plugins/openlayers/layers/openlayers-geoJson-layers.plugin.ts [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+		map((map: CaseMapState) => map.flags.layers),
+		distinctUntilChanged()
+	);
+
+	update$ = combineLatest(this.store$.select(selectLayers), this.store$.select(selectSelectedLayersIds), this.isHidden$)
+		.pipe(
+			mergeMap(([result, selectedLayerId, isHidden]: [ILayer[], string[], boolean]) => {
+				const obsList = result
+					.filter((layer: ILayer) => layer.layerPluginType === layerPluginType.geoJson)
+					.map((layer: ILayer) => {
```
[the remark this answers is not in the document]
Comment (Blaussh) [reply src/app/@ansyn/plugins/openlayers/layers/openlayers-geoJson-layers.plugin.ts]: changed the implementation by taking the if/else to a fanction

File: src/app/@ansyn/plugins/openlayers/layers/openlayers-osm-layers.plugin.ts [hunk 6]
```diff
@@ -0,0 +1,100 @@
...
+				crossOrigin: null
+			})
+		});
+		vector.set('id', layer.id);
+		return vector;
+	}
+
+	addGroupLayer(layer: ILayer) {
+		const group = OpenLayersMap.groupLayers.get('layers');
+		if (!group.getLayers().getArray().some((shownLayer) => shownLayer.get('id') === layer.id)) {
```
[the remark this answers is not in the document]
Comment (Blaussh) [reply src/app/@ansyn/plugins/openlayers/layers/openlayers-osm-layers.plugin.ts]: 👍

File: src/app/@ansyn/plugins/openlayers/layers/openlayers-osm-layers.visualizer.ts [hunk 7]
```diff
@@ -18,12 +18,12 @@ import { ImageryPlugin } from '@ansyn/imagery/model/decorators/imagery-plugin';
 	supported: [OpenLayersMap],
 	deps: [Store]
 })
-export class OpenlayersOsmLayersPlugin extends BaseImageryPlugin {
+export class OpenlayersOsmLayersVisualizer extends BaseImageryPlugin {
```
Comment (asafMasa) [inline src/app/@ansyn/plugins/openlayers/layers/openlayers-osm-layers.visualizer.ts]: rename back to OpenlayersOsmLayersPlugin

**Pull request merged**